src/send-bsd.h

```diff
@@ -34,4 +35,48 @@ int send_packet(sock_t sock, void *buf, int len, UNUSED uint32_t idx)
 	return write(sock.sock, buf, len);
 }
 
+// Since BSD doesn't have the sendmmsg syscall leveraged in send-linux.c, this just wraps the single send_packet call.
+// However, the behavior in sendmmsg is to send as many packets as possible until one fails, and then return the number of sent packets.
+// Following the same pattern for consistency
+// Returns - number of packets sent
+// Returns last error code if no packets could be sent successfully
+int send_batch(sock_t sock, batch_t* batch, int retries) {
+	int packets_sent = 0;
+	int rc = 0;
+	for (int i=0;i<batch->len;i++) {
+		int retry_ct = 0;
+		for (retry_ct = 0; retry_ct < retries; retry_ct++) {
+			rc = send_packet(sock, ((void *)batch->packets) + (i * MAX_PACKET_SIZE), batch->lens[i], 0);
+			if (rc >= 0) {
+				packets_sent++;
+				break;
+			}
+		}
+		if (rc < 0) {
+			// packet couldn't be sent in retries number of attempts
+			struct in_addr addr;
+			addr.s_addr = batch->ips[i];
+			char addr_str_buf
+			    [INET_ADDRSTRLEN];
+			const char *addr_str =
+			    inet_ntop(
+				AF_INET, &addr,
+				addr_str_buf,
+				INET_ADDRSTRLEN);
+			if (addr_str != NULL) {
+				log_debug(
+				    "send",
+				    "send_packet failed for %s. %s",
+				    addr_str,
```

Comment:
Can you fix up indentation here — pretty sure `errno` stuff isn't pushing us past 80char world